make/go.mk

```diff
@@ -25,7 +25,6 @@ define go_test
 		-- \
 		-covermode=atomic \
 		-coverprofile=coverage.out \
-		-race \
```

Comment:
Why is this being removed?

Reply:
Because there's a race in controller-runtime envtest... I noticed that CAPA, which uses this same approach, also doesn't enable race here.